src/flow/src/adapter/worker.rs

```diff
@@ -0,0 +1,508 @@
+                ),
+            }
+            .fail()
+            .with_context(|_| EvalSnafu {})
+        }
+    }
+
+    /// remove task, return task id
+    pub async fn remove_flow(&self, task_id: FlowId) -> Result<bool, Error> {
+        let req = Request::Remove { task_id };
+        let ret = self.itc_client.lock().await.call_blocking(req).await?;
+        if let Response::Remove { result } = ret {
+            Ok(result)
+        } else {
+            InternalSnafu {
+                reason: format!("Flow Node/Worker failed, expect Response::Remove, found {ret:?}"),
+            }
+            .fail()
+            .with_context(|_| EvalSnafu {})
+        }
+    }
+
+    /// trigger running the worker, will not block, and will run the worker parallelly
+    ///
+    /// will set the current timestamp to `now` for all dataflows before running them
+    pub async fn run_available(&self, now: repr::Timestamp) {
+        self.itc_client
+            .lock()
+            .await
+            .call_non_blocking(Request::RunAvail { now })
+            .await;
+    }
+
+    pub async fn contains_flow(&self, task_id: FlowId) -> Result<bool, Error> {
+        let req = Request::ContainTask { task_id };
+        let ret = self
+            .itc_client
+            .lock()
+            .await
+            .call_blocking(req)
+            .await
+            .unwrap();
+        if let Response::ContainTask {
+            result: task_contain_result,
+        } = ret
+        {
+            Ok(task_contain_result)
+        } else {
+            InternalSnafu {
+                reason: format!(
+                    "Flow Node/Worker itc failed, expect Response::ContainTask, found {ret:?}"
+                ),
+            }
+            .fail()
+            .with_context(|_| EvalSnafu {})
+        }
+    }
+
+    /// shutdown the worker
+    pub async fn shutdown(&self) {
+        self.itc_client
+            .lock()
+            .await
+            .call_non_blocking(Request::Shutdown)
+            .await;
+    }
+}
+
+/// The actual worker that does the work and contain active state
+#[derive(Debug)]
+pub struct Worker<'subgraph> {
+    /// Task states
+    pub(crate) task_states: BTreeMap<FlowId, ActiveDataflowState<'subgraph>>,
+    itc_server: Arc<Mutex<InterThreadCallServer>>,
+}
+
+impl<'s> Worker<'s> {
+    #[allow(clippy::too_many_arguments)]
+    pub fn create_flow(
+        &mut self,
+        task_id: FlowId,
+        plan: TypedPlan,
+        sink_id: GlobalId,
+        sink_sender: mpsc::UnboundedSender<DiffRow>,
+        source_ids: &[GlobalId],
+        src_recvs: Vec<broadcast::Receiver<DiffRow>>,
+        // TODO(discord9): set expire duration for all arrangement and compare to sys timestamp instead
+        expire_when: Option<repr::Duration>,
+        create_if_not_exist: bool,
+        err_collector: ErrCollector,
+    ) -> Result<Option<FlowId>, Error> {
+        let _ = expire_when;
+        if create_if_not_exist {
```

Comment:
Should an error be reported when `create_if_not_exist == false` and `task_id` already exist?